code/modules/fish/fishtank.dm

```diff
@@ -0,0 +1,690 @@
+	else if(filth_level == 10)
+		examine_message += "[src] is absolutely disgusting! Someone should clean it NOW! "
+
+	examine_message += "<br>Food level: "
+
+	//Approximate food level
+	if(!fish_count)								//Check if there are fish in the tank
+		if(food_level > 0)						//Don't report a tank that has neither fish nor food in it
+			examine_message += "There's some food in [src], but no fish! "
+	else										//We've got fish, report the food level
+		if(food_level == 0)
+			examine_message += "The fish look very hungry! "
+		else if(food_level < 2)
+			examine_message += "The fish are nibbling on the last of their food. "
+		else if(food_level < 10)				//Breeding is possible
+			examine_message += "The fish seem happy! "
+		else if(food_level == 10)
+			examine_message += "There is a solid layer of fish food at the top. "
+
+	//Report the number of harvestable eggs
+	if(egg_count)								//Don't bother if there isn't any eggs
+		examine_message += "<br>There are [egg_count] eggs able to be harvested! "
+
+	examine_message += "<br>"
+
+	//Report the number and types of live fish if there is water in the tank
+	if(fish_count == 0)
+		examine_message += "[src] doesn't contain any live fish. "
+	else
+		//Build a message reporting the types of fish
+		var/fish_num = fish_count
+		var/message = "You spot "
+		while(fish_num > 0)
+			var/datum/fish/fish_type = fish_list[fish_num]
+			var/fish_name = fish_type.fish_name
+			if(fish_count > 1 && fish_num == 1)	//If there were at least 2 fish, and this is the last one, add "and" to the message
+				message += "and "
+			message += "\an [fish_name]"
+			fish_num --
+			if(fish_num > 0)					//There's more fish, add a comma to the message
+				message +=", "
+		message +="."							//No more fish, end the message with a period
+		//Display the number of fish and previously constructed message
+		examine_message += "[src] contains [fish_count] live fish. [message] "
+
+	examine_message += "<br>"
+
+	//Report lid state for tanks and wall-tanks
+	if(has_lid)									//Only report if the tank actually has a lid
+		//Report lid state
+		if(lid_switch)
+			examine_message += "The lid is closed. "
+		else
+			examine_message += "The lid is open. "
+
+	examine_message += "<br>"
+
+	//Report if the tank is leaking/cracked
+	if(water_level)							//Tank has water, so it's actually leaking
+		if(leaking == 1)
+			examine_message += "[src] is leaking."
+		if(leaking == 2)
+			examine_message += "[src] is leaking profusely!"
+	else										//No water, report the cracks instead
+		if(leaking == 1)
+			examine_message += "[src] is cracked."
+		if(leaking == 2)
+			examine_message += "[src] is nearly shattered!"
+
+
+	//Finally, report the full examine_message constructed from the above reports
+	to_chat(user, "[examine_message]")
+	return examine_message
+
+//////////////////////////////
+//		ATTACK PROCS		//
+//////////////////////////////
+
+/*
+/obj/machinery/fishtank/UnarmedAttack(mob/living/simple_animal/M)
```

Comment:
I don't particularly see the need for this, and with the simple_mob rewrite currently being worked on, this would just make more work for neerti to try and add support for this behaviour

Reply:
I didn't know what to do with this code and i didn't want it trashed and us never getting that feature so it's here in wait. neerti is absolutely not obligated to support that.